packages/block-library/src/image/edit.native.js

```diff
@@ -448,7 +448,9 @@ export class ImageEdit extends Component {
 			imageDefaultSize,
 		] );
 
-		const isFeaturedImage = featuredImageId === attributes.id;
+		// Disabled on iOS until https://github.com/WordPress/gutenberg/pull/31345 is merged in to provide iOS bridge support.
+		const isFeaturedImage =
```

Comment:
+1 for a fix based on checking for undefined, that would also cover the same case on Android.

While at it, I think we should check for undefined directly instead of the plain `featuredImageId ? ...` in the suggestion Siobhan. If featuredImageId is `0` it will check as falsy but it might be a valid image id. So, wdyt about using:
`const isFeaturedImage = (typeof featuredImageId !== 'undefined') && (featuredImageId === attributes.id)`?

Reply:
Implemented here 9583973